File: src/main/java/net/minestom/server/listener/ResourcePackListener.java
package net.minestom.server.listener;

import net.minestom.server.entity.Player;
import net.minestom.server.event.EventDispatcher;
import net.minestom.server.event.player.PlayerResourcePackStatusEvent;
import net.minestom.server.network.packet.client.play.ClientResourcePackStatusPacket;

public class ResourcePackListener {

    public static void listener(ClientResourcePackStatusPacket packet, Player player) {
        EventDispatcher.call(new PlayerResourcePackStatusEvent(player, packet.status()));
    }
}

File: src/main/java/net/minestom/server/listener/ResourcePackListener.kt
package net.minestom.server.listener

import net.minestom.server.entity.Player
import net.minestom.server.event.EventDispatcher
import net.minestom.server.network.packet.client.play.ClientResourcePackStatusPacket
import net.minestom.server.event.player.PlayerResourcePackStatusEvent

object ResourcePackListener {
    @JvmStatic
    fun listener(packet: ClientResourcePackStatusPacket, player: Player?) {
        EventDispatcher.call(PlayerResourcePackStatusEvent(player!!, packet.status()))
    }
}